File: .devin/skills/cadence-workflow-orchestration/examples/python/hello_world_workflow.py
"""Minimal hello-world Cadence workflow + activity for Python.

See ../../00-get-started/local-quickstart.md for how to run a local server and domain,
and ../../03-python-client/workflows-and-activities.md for the full explanation of every
piece here.
"""

from datetime import timedelta

from cadence import Registry, activity, workflow
from cadence.workflow import execute_activity

registry = Registry()


@activity.defn()
async def hello_world_activity(name: str) -> str:
    activity.info()  # demonstrates activity context access; see workflows-and-activities.md
    return f"Hello {name}!"


@registry.workflow()
class HelloWorldWorkflow:
    @workflow.run
    async def run(self, name: str) -> str:
        return await execute_activity(
            "hello_world_activity",
            str,
            name,
            start_to_close_timeout=timedelta(minutes=1),
            heartbeat_timeout=timedelta(seconds=20),
        )

File: .devin/skills/cadence-workflow-orchestration/examples/go/hello_world_workflow.go
// Minimal hello-world Cadence workflow + activity for Go.
// See ../../00-get-started/local-quickstart.md for how to run a local server and domain,
// and ../../02-go-client/workflows-and-workers.md for the full explanation of every piece here.
package helloworld

import (
	"context"
	"time"

	"go.uber.org/cadence/activity"
	"go.uber.org/cadence/workflow"
)

func init() {
	workflow.Register(HelloWorldWorkflow)
	activity.Register(HelloWorldActivity)
}

// HelloWorldWorkflow orchestrates a single activity call.
func HelloWorldWorkflow(ctx workflow.Context, name string) (string, error) {
	ao := workflow.ActivityOptions{
		ScheduleToStartTimeout: time.Minute,
		StartToCloseTimeout:    time.Minute,
		HeartbeatTimeout:       time.Second * 20,
	}
	ctx = workflow.WithActivityOptions(ctx, ao)

	logger := workflow.GetLogger(ctx)
	logger.Info("HelloWorldWorkflow started")

	var result string
	err := workflow.ExecuteActivity(ctx, HelloWorldActivity, name).Get(ctx, &result)
	if err != nil {
		logger.Error("Activity failed.")
		return "", err
	}

	logger.Info("HelloWorldWorkflow completed")
	return result, nil
}

// HelloWorldActivity performs the actual (non-deterministic-safe) work.
func HelloWorldActivity(ctx context.Context, name string) (string, error) {
	activity.GetLogger(ctx).Info("HelloWorldActivity called")
	return "Hello " + name + "!", nil
}
